lib/build.gradle

```diff
@@ -514,26 +513,31 @@ def setupJNILibraryBuildTasks(library, platform) {
 def createCopyNativeLibraryTask(library, targetPlatform) {
     if (CBL_NATIVE_LIBRARIES.contains(library)) return
 
-    CBL_NATIVE_LIBRARIES.add(library)
-    
-    def libs = ["macos/x86_64/libLiteCore.dylib", 
-                "linux/x86_64/libLiteCore.so", 
-                "windows/x86_64/LiteCore.dll"]
-    for (lib in libs) {
-        def libFile = file("${CBL_CORE_NATIVE_DIR}/${lib}")
-        if (libFile.exists()) {
-            CBL_NATIVE_LIBRARIES.add(libFile)
+    def libPath = getNativeLibraryResourcePath(targetPlatform)
+    task copyJni(type: Copy, dependsOn: LiteCoreJNISharedLibrary) {
+        from library
+        into "${CBL_NATIVE_DIR}/libs/${libPath}"
+    }
+
+    task copyLiteCore(type: Copy, dependsOn: LiteCoreJNISharedLibrary) {
+        from ("${CBL_CORE_NATIVE_DIR}") {
+            include "macos/**"
+            include "linux/**"
+            include "windows/**"
+            exclude "**/*.a"
         }
+        into "${CBL_NATIVE_DIR}/libs"
     }
 
-    def libPath = getNativeLibraryResourcePath(targetPlatform)
-    def task = tasks.create("copyNativeLibraries", Copy) {
-        from CBL_NATIVE_LIBRARIES
-        into "${CBL_NATIVE_DIR}/libs/${libPath}"
+    task generateNativeLibraryMD5(dependsOn: ["copyJni", "copyLiteCore"]) {
+        doFirst {
+            fileTree(dir: "${CBL_NATIVE_DIR}/libs").each { File file ->
+                ant.checksum(file: file)
+            }
+        }
     }
 
-    processResources.dependsOn(task)
-    task.dependsOn(LiteCoreJNISharedLibrary)
+    processResources.dependsOn(["generateNativeLibraryMD5"])
 }
```

Comment:
I thought there was discussion of putting the lib into a directory named for its hash.  Did that not work?

Reply:
That is in another PR when extracting.

Reply:
cool.  got it.